tests/transforms/test_split_to_single_terms.py

```diff
@@ -0,0 +1,661 @@
+        assert len(tapes) == 1
+        assert tapes[0].measurements == [
+            qml.expval(qml.X(0)),
+            qml.expval(qml.X(1)),
+            qml.expval(qml.Y(1)),
+        ]
+        assert fn(([0.1, 0.2, 0.3],)) == (0.1 + 0.1, 0.2 + 0.3 + 0.3)
+
+    @pytest.mark.parametrize("batch_type", (tuple, list))
+    def test_batch_of_tapes(self, batch_type):
+        """Test that `split_to_single_terms` can transform a batch of tapes with multi-term observables"""
+
+        tape_batch = [
+            qml.tape.QuantumScript([qml.RX(1.2, 0)], [qml.expval(qml.X(0) + qml.Y(0) + qml.X(1))]),
+            qml.tape.QuantumScript([qml.RY(0.5, 0)], [qml.expval(qml.Z(0) + qml.Y(0))]),
+        ]
+        tape_batch = batch_type(tape_batch)
+
+        tapes, fn = split_to_single_terms(tape_batch)
+
+        expected_tapes = [
+            qml.tape.QuantumScript(
+                [qml.RX(1.2, 0)],
+                [qml.expval(qml.X(0)), qml.expval(qml.Y(0)), qml.expval(qml.X(1))],
+            ),
+            qml.tape.QuantumScript([qml.RY(0.5, 0)], [qml.expval(qml.Z(0)), qml.expval(qml.Y(0))]),
+        ]
+        for actual_tape, expected_tape in zip(tapes, expected_tapes):
+            assert qml.equal(actual_tape, expected_tape)
+
+        result = ([0.1, 0.2, 0.3], [0.4, 0.2])
+        assert fn(result) == ((0.1 + 0.2 + 0.3), (0.4 + 0.2))
+
+    @pytest.mark.parametrize(
+        "non_pauli_obs", [qml.Projector([0], wires=[1]), qml.Hadamard(wires=[1])]
+    )
+    def test_tape_with_non_pauli_obs(self, non_pauli_obs):
+        """Tests that the tape is split correctly when containing non-Pauli observables"""
+
+        measurements = [
+            qml.expval(qml.X(0) + qml.Y(0) + qml.X(1)),
+            qml.expval(non_pauli_obs + qml.Z(3)),
+        ]
+        tape = qml.tape.QuantumScript([qml.RX(1.2, 0)], measurements=measurements)
+
+        tapes, fn = split_to_single_terms(tape)
+
+        expected_tape = qml.tape.QuantumScript(
+            [qml.RX(1.2, 0)],
+            [
+                qml.expval(qml.X(0)),
+                qml.expval(qml.Y(0)),
+                qml.expval(qml.X(1)),
+                qml.expval(non_pauli_obs),
+                qml.expval(qml.Z(3)),
+            ],
+        )
+
+        assert qml.equal(tapes[0], expected_tape)
+
+        result = [[0.1, 0.2, 0.3, 0.4, 0.5]]
+        assert fn(result) == ((0.1 + 0.2 + 0.3), (0.4 + 0.5))
+
+    @pytest.mark.parametrize(
+        "observable",
+        [
+            qml.X(0) + qml.Y(1),
+            2 * (qml.X(0) + qml.Y(1)),
+            3 * (2 * (qml.X(0) + qml.Y(1)) + qml.X(1)),
+        ],
+    )
+    def test_splitting_sums_in_unsupported_mps_raises_error(self, observable):
+
+        tape = qml.tape.QuantumScript([qml.X(0)], measurements=[qml.counts(observable)])
+        with pytest.raises(
+            RuntimeError, match="Cannot split up terms in sums for MeasurementProcess"
+        ):
+            _, _ = split_to_single_terms(tape)
+
+
+@patch(
```

Comment:
```suggestion
# this patch can be rewritten and updated if the details of default_qubit.observables change
# testing hack and not indicative of a dependency or coupling
@patch(
```


Or maybe we could do something like:
```
@qml.devices.modifiers.single_tape_support
class NoTermsDevice(qml.devices.DefaultQubit):

    def execute(self, circuits, execution_config = qml.devices.DefaultExecutionConfig):
        for t in circuits:
            for mp in t.measurements:
               if mp.obs and len(mp.obs.terms()[0]) > 1:
                   raise ValueError("no terms device does not accept observables with multiple terms")
        return super().execute(circuits, execution_config)
```

And then use that device?

Reply:
added a custom device instead of using the patch!